modules/axhal/src/lib.rs

```diff
@@ -124,3 +126,76 @@ pub fn init_early(cpu_id: usize, arg: usize) {
 pub fn get_bootarg() -> usize {
     *BOOT_ARG
 }
+
+/// The number of CPUs in the system. Based on the number declared by the
+/// platform crate and limited by the configured maximum CPU number.
+///
+/// Defaults to 1 for non-SMP builds.
+#[cfg(feature = "smp")]
+static CPU_NUM: AtomicUsize = AtomicUsize::new(1);
+
+/// Gets the number of CPUs running in the system.
+///
+/// When SMP is disabled, this function always returns 1.
+///
+/// When SMP is enabled, It's the smaller one between the platform-declared CPU
+/// number [`axplat::power::cpu_num`] and the configured maximum CPU number
+/// `axconfig::plat::MAX_CPU_NUM`.
+///
+/// This value is determined during the BSP initialization phase.
+pub fn cpu_num() -> usize {
```

Comment:
Use `Once` or `lazy_static` to avoid manually calling the initialization function?